ros/api/v1/hosts.py

```diff
@@ -466,6 +501,31 @@ def get(self):
         io_states_dict['undersized'] = -1
         total_conditions = totals['cpu'] + totals['memory'] + totals['io']
 
+        current_utc_datetime = datetime.now(timezone.utc)
+        upload_date = (current_utc_datetime - timedelta(days=1)).date()
+
+        current_performance_profiles = [
+            record
+            for record in systems_with_performance_record_queryset
+            if record.report_date.date() == upload_date
+        ]
+
+        aliased_performance_profile_history = aliased(PerformanceProfileHistory, name='performance_profile_history')
+
+        historical_performance_profiles = db.session.query(
+            systems_with_performance_record_subquery.c.system_id,
+            systems_with_performance_record_subquery.c.rule_hit_details
+        ).select_from(
+            aliased_performance_profile_history
+        ).join(
+            systems_with_performance_record_subquery,
+            systems_with_performance_record_subquery.c.system_id == aliased_performance_profile_history.system_id
+        )  # Systems older than 7 days/stale systems are considered here
```

Comment:
@saltgen This is too complicated and also returning incorrect results.
If we just want to select the systems from `PerformanceProfileHistory` where `system_id = systems_with_performance_record_subquery.c.system_id`. Then please replace it with the below. 
```suggestion
       historical_performance_profiles = db.session.query(PerformanceProfileHistory).join(
            systems_with_performance_record_subquery,
            systems_with_performance_record_subquery.c.system_id == PerformanceProfileHistory.system_id
        ) # Systems older than 7 days/stale systems are considered here
```

Reply:
@patilsuraj767 Yeah this is cleaner I agree, just that I have added changes to select only certain columns for `PerformanceProfileHistory` so that entire records need not get loaded into memory.

Change pushed